controllers/provisioning_controller.go

```diff
@@ -200,14 +204,6 @@ func getSuccessStatus(imageCacheState appsv1.DaemonSetConditionType, ironicProxy
 // Reconcile updates the cluster settings when the Provisioning
 // resource changes
 func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// provisioning.metal3.io is a singleton
-	// Note: this check is here to make sure that the early startup configuration
-	// is correct. For day 2 operatations the webhook will validate this.
-	if req.Name != metal3iov1alpha1.ProvisioningSingletonName {
-		klog.Info("ignoring invalid CR", "name", req.Name)
-		return ctrl.Result{}, nil
-	}
-
```

Comment:
I replaced this with a predicate (see method `SetupWithManager`), so only the expected resource will trigger.

Reply:
The current version has a benefit of logging what went wrong though.